File: May_2022/02_05_2022/goto.py
n = int(input())
deadline = [list(map(int,input().split())) for i in range(n)]
m = int(input())
worked = [list(map(int,input().split())) for i in range(m)]

half = 0
if(n%2 == 0):
    half = n/2
else:
    half = (n+1)/2

final = 0

for i in deadline:
    flag = True
    count = 0
    for j in worked:
        if(j[0]<=i[0]):
            count += j[1]
    if(count>=i[1]):
        final += 1

if(final>=half):
    print("Allowed")
else:
    print("Denied")

File: May_2022/02_05_2022/goto.cpp
#include <iostream>
#include <vector>

using namespace std;
 
int main(){

    int n,m;
    vector<vector<int>> deadline,worked;

    cin>>n;

    for(int i=0;i<n;i++){
        int a,b;
        cin>>a>>b;
        vector<int> temp = {a,b};
        deadline.push_back(temp);
    }

    cin>>m;

    for(int i=0;i<m;i++){
        int a,b;
        cin>>a>>b;
        vector<int> temp = {a,b};
        worked.push_back(temp);
    }

    int final = 0;
    int half;

    if(n%2 == 0){
        half = n/2;
    }
    else{
        half = (n+1)/2;
    }

    for(int i=0;i<n;i++){
        bool flag = true;
        int count = 0;
        for(int j=0;j<m;j++){
            if(deadline[i][0]>=worked[j][0]){
                count += worked[j][1];
            }
        }

        if(count>=deadline[i][1]){
            final++;
        }
    }

    if(final>=half){
        cout<<"Allowed";
    }
    else{
        cout<<"Denied";
    }

    return 0;
}